src/libsystemd-network/sd-radv.c

```diff
@@ -659,15 +654,18 @@ int sd_radv_add_prefix(sd_radv *ra, sd_radv_prefix *p) {
         return 0;
 }
 
-sd_radv_prefix *sd_radv_remove_prefix(sd_radv *ra,
-                                               const struct in6_addr *prefix,
-                                               unsigned char prefixlen) {
-        sd_radv_prefix *cur, *next;
+void sd_radv_remove_prefix(
```

Comment:
Could you please split this function refactor in a separate commit? Lumped together with the tree-wide iterator change it's very easy to miss

Reply:
Done.